### PR TITLE
[Serialization] Drop support for serializing LValueType.

### PR DESCRIPTION
**Pull request by jrose-apple**
This type never appears in API boundaries or in SIL instructions, which means it never gets serialized.

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
@swift-ci Please test macOS

**Comment by jrose-apple** [comment]
@swift-ci Please test macOS

File: lib/Serialization/Deserialization.cpp [hunk 1]
```diff
@@ -3986,15 +3986,9 @@ Expected<Type> ModuleFile::getTypeChecked(TypeID TID) {
...
-  case decls_block::LVALUE_TYPE: {
-    TypeID objectTypeID;
-    decls_block::LValueTypeLayout::readRecord(scratch, objectTypeID);
-    typeOrOffset = LValueType::get(getType(objectTypeID));
-    break;
-  }
   case decls_block::INOUT_TYPE: {
     TypeID objectTypeID;
-    decls_block::LValueTypeLayout::readRecord(scratch, objectTypeID);
+    decls_block::InOutTypeLayout::readRecord(scratch, objectTypeID);
```
Comment (slavapestov) [inline lib/Serialization/Deserialization.cpp]: Before, we were serializing INOUT_TYPE using LValueTypeLayout. Is this what's causing the test failure here?

Comment (jrose-apple) [inline lib/Serialization/Deserialization.cpp]: Nope, the test failure is completely unrelated. There's unfortunately no static or dynamic type-checking on the layout deserialization, although maybe there should be.

**Comment by jrose-apple** [comment]
@swift-ci Please test macOS

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test Linux

**Pull request merged**